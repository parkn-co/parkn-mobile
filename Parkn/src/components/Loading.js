import React, {Component} from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Animated,
  Easing,
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {bluePalette, grayPalette} from '../styles/colors';
import {fontNames} from '../styles/fonts';

const iconColorsByType = {
  default: 'rgba(255,255,255,0.75)',
  light: bluePalette.medium,
}

export default class Loading extends Component {
  constructor(props) {
    super(props);

    this.state = {
      animatedValue: new Animated.Value(0),
      toValue: 100,
    };

    this.startRotation = this.startRotation.bind(this);
  }

  componentDidMount() {
    this.startRotation();

    this.interval = setInterval(this.startRotation, 3000);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  startRotation() {
    Animated.timing(this.state.animatedValue, {
      toValue: this.state.toValue,
      duration: 3000,
      easing: Easing.linear
    }).start();

    this.setState({toValue: this.state.toValue + 100});
  }

  render() {
    const rotate = this.rotateValue;
    const type = this.props.type || 'default';
    
    return (
      <View style={[styles.container, styles[type]]}>
        <Animated.View
          style={{transform: [{rotate}]}}
        >
          <Icon name="cached" size={50} color={iconColorsByType[type]} />
        </Animated.View>
      </View>
    );
  }

  get rotateValue() {
    return this.state.animatedValue.interpolate({
      inputRange: [0, 100],
      outputRange: ['0deg', '360deg']
    });
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  default: {
    backgroundColor: bluePalette.medium,
  },
  light: {
    backgroundColor: '#F5FCFF',
  }
});
