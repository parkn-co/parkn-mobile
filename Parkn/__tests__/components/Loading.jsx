/* eslint-env jest */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Loading from '../../src/components/Loading';

describe('Loading', () => {
  it('renders correctly', () => {
    expect(renderer.create(
      <Loading />
    )).toMatchSnapshot();
  });
});
