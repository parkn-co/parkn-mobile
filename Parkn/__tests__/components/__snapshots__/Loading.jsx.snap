exports[`Loading renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      },
      Object {
        "backgroundColor": "#076FEB",
      },
    ]
  }>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "rotate": "0deg",
          },
        ],
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(255,255,255,0.75)",
            "fontFamily": "Material Icons",
            "fontSize": 50,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }>
      î¡ª
    </Text>
  </View>
</View>
`;
